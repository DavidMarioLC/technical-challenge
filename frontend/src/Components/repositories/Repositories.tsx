import './repositories.css';
import Repository from '../repository/Repository';
import {IRepository} from '../../interfaces/repository.interface';

interface Props{
  repositories:IRepository[]
}


const Repositories = ({repositories}: Props) => {
  return (
    <div className="repositories"> 
      {
        repositories.map((repo,index)=>(
          <Repository 
             key={index}
             name={repo.name} 
             description={repo.description}
             language={repo.language}
             visibility={repo.visibility}
             owner={repo.owner}
             pushed_at={repo.pushed_at}
             license={repo.license}
             stargazers_count={repo.stargazers_count}
             />
        ))

      }
    </div>
  )
}

export default Repositories
