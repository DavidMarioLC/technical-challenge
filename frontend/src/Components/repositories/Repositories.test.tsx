import renderer from 'react-test-renderer';
import Repositories from './Repositories';


it('render Repositories correctly', () => {
  const tree = renderer.create(<Repositories repositories={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});