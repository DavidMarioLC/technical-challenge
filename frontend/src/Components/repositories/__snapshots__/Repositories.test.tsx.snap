// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Repositories correctly 1`] = `
<div
  className="repositories"
/>
`;
